python/cudf/cudf/tests/test_concat.py

```diff
@@ -1997,3 +1997,38 @@ def test_concat_dict_incorrect_type_index(d):
         match="cannot concatenate a dictionary containing indices",
     ):
         cudf.concat(d, axis=1)
+
+
+@pytest.mark.parametrize(
+    "axis",
+    [
+        0,
+        pytest.param(
+            1,
+            marks=pytest.mark.xfail(
+                reason=("cannot concatenate indices across axis 1")
+            ),
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "idx",
+    [
+        [cudf.Index([1, 2, 3])],
+        [cudf.Index([1, 2, 3]), cudf.Index([4, 5, 6])],
+        [
+            cudf.MultiIndex(
+                levels=[[1, 2], ["blue", "red"]],
+                codes=[[0, 0, 1, 1], [1, 0, 1, 0]],
+            )
+        ],
+        [cudf.CategoricalIndex([1, 2, 3])],
```

Comment:
*suggestion*: As previously, please don't make the objects in the fixture parameter list, either hide them behind a lambda, or use a `@pytest.fixture` (I prefer this, though it's a bit more typing).

To avoid too many name collisions in fixtures, I like (although the test suite doesn't do a lot of it) to introduce classes:

```
class TestConcat:
    @pytest.fixture(params=["single-index", "multiple-indices", "multiindex", "categorical-index"])
    def index(self, request):
        if request.param == "single-index":
            return [cudf.Index(...)]
        ...

     @pytest.mark.parametrize("axis,exception", ...) # as above
     def test_concat_index(self, index, axis, exception):
         with pytest.raises(exception):
              result = cudf.concat(index, axis=axis)
         assert ...
```

Bit fiddly, if we raise an exception then result is not assigned so the rest of the test body can't run, which is a bit annoying.

Reply:
yea, can do. fyi i'm fixing all the instantiations in a different PR